resolwe_bio/processes/reads_processing/bbduk.yml

```diff
@@ -236,464 +378,684 @@
+        mincovfraction={{ processing.min_coverage_fraction }} \
+        hammingdistance={{ processing.hamming_distance }} \
+        qhdist={{ processing.query_hamming_distance }} \
+        editdistance={{ processing.edit_distance }} \
+        hammingdistance2={{ processing.hamming_distance2 }} \
+        qhdist2={{ processing.query_hamming_distance2 }} \
+        editdistance2={{ processing.edit_distance2 }} \
+        forbidn={{ bool[processing.forbid_N] }} \
+        findbestmatch={{ bool[processing.find_best_match] }} \
+        {% if operations.k_trim != 'f' %} \
+          ktrim={{ operations.k_trim }} \
+        {% endif %} \
+        {% if operations.k_mask != 'f' and operations.k_mask != 't' %} \
+          kmask={{ operations.k_mask }} \
+        {% endif %} \
+        maskfullycovered={{ bool[operations.mask_fully_covered] }} \
+        mink={{ operations.min_k }} \
+        {% if operations.quality_trim != 'f' %} \
+          qtrim={{ operations.quality_trim }} \
+        {% endif %} \
+        trimq={{ operations.trim_quality }} \
+        trimpolya={{ operations.trim_poly_A }} \
+        minlength={{ operations.min_length }} \
+        minlengthfraction={{ operations.min_length_fraction }} \
+        {% if operations.max_length %} \
+          maxlength={{ operations.max_length }} \
+        {% endif %} \
+        minavgquality={{ operations.min_average_quality }} \
+        maqb={{ operations.min_average_quality_bases }} \
+        minbasequality={{ operations.min_base_quality }} \
+        minconsecutivebases={{ operations.min_consecutive_bases }} \
+        trimpad={{ operations.trim_pad }} \
+        minoverlap={{ operations.min_overlap }} \
+        mininsert={{ operations.min_insert }} \
+        forcetrimleft={{ operations.force_trim_left }} \
+        forcetrimright={{ operations.force_trim_right }} \
+        forcetrimright2={{ operations.force_trim_right2 }} \
+        forcetrimmod={{ operations.force_trim_mod }} \
+        restrictleft={{ operations.restrict_left }} \
+        restrictright={{ operations.restrict_right }} \
+        mingc={{ operations.min_GC }} \
+        maxgc={{ operations.max_GC }} \
+        tossjunk={{ bool[operations.toss_junk] }} \
+        chastityfilter={{ bool[header_parsing.chastity_filter] }} \
+        barcodefilter={{ bool[header_parsing.barcode_filter] }} \
+        {% if barcodes %} \
+          barcodes={{ barcodes|join(',') }} \
+        {% endif %} \
+        xmin={{ header_parsing.x_min }} \
+        ymin={{ header_parsing.y_min }} \
+        xmax={{ header_parsing.x_max }} \
+        ymax={{ header_parsing.y_max }} \
+        entropy={{ complexity.entropy }} \
+        entropywindow={{ complexity.entropy_window }} \
+        entropyk={{ complexity.entropy_k }} \
+        minbasefrequency={{ complexity.min_base_frequency }} \
+        entropymask={{ bool[complexity.entropy_mask] }} \
+        threads={{ requirements.resources.cores }} \
         -Xmx1g
```

Comment:
Its better to have the memory in requirements resources and than use jinja formating like [here](https://github.com/genialis/resolwe-bio/pull/461/files#diff-50334eb3c7235ebcce15599e56155610R179)

Reply:
The idea is interesting but the implementation is poor. See [this RedHat post](https://developers.redhat.com/blog/2017/04/04/openjdk-and-containers/):
> Specifying -Xmx=1g is telling the JVM to allocate a 1gb heap. It’s not telling the JVM to limit its entire memory usage to 1gb. There are card tables, code caches, and all sorts of other off heap data structures.

Even by specifying `-XX:MaxRAM` I don't think that it's wise to instruct the tool to use all RAM in `requirements.resources.memory` up to the very last byte. There is likely some overhead by the shell etc. as well. @tjanez may want to contribute here.

Reply:
I based the maximum memory setting on the suggestion in [the official documentation](https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/bbduk-guide/), paragraph _Memory_. I like the idea of making the memory limit used by Java dependent on `requirements.resources.memory`. Thanks for the suggestion.

Reply:
A bit naive, but you could assign 2GB (or 1.5GB, ..) using `requirements.resources.memory`, and use only portion of that in the java flag via division/rounding. This would be better than the defaults that are assigned to a whole process now (4 or 8 GB?)

Reply:
I've set `-Xmx` to 85% of memory specified in requirements, as suggested in the help of BBDuk:
> The max is typically 85% of physical memory.